compiler-core/src/build/project_compiler.rs

```diff
@@ -122,6 +125,8 @@ where
         }
         let result = self.compile_root_package();
 
+        self.check_build_journal();
```

Comment:
Let's check for errors here
```suggestion
        self.check_build_journal()?;
```